src/context/client-context.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, createContext, useState, useEffect } from "react";
+import setupAxios from "../common/axios";
+import qdrantClient from "../common/client";
+
+const DEFAULT_SETTINGS = {
+  apiKey: "",
+};
+
+//Write settings to local storage
+const persistSettings = (settings) => {
+  localStorage.setItem("settings", JSON.stringify(settings));
```

Comment:
To prevent juggling between JS-JSON you _could_ also use [indexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) as it stores objects. But that is definitely a bit harder to work with.

Reply:
I would prefer to keep it simple for now, this isn't something which would require backward compatibility